### PR TITLE
Modify Openhim core api package to fetch core url from default.json file

### PR DESCRIPTION
**Pull request by drizzentic**

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86bywx2j0 SSO isn't enabled on the OpenHIM console anymore when enabled via env vars](https://app.clickup.com/t/86bywx2j0)

**Review by rcrichton — CHANGES_REQUESTED** [review]
Looks good, a few comments.

Also we can remove the old Dockerfile in the root and we can also remove this copied file - https://github.com/jembi/openhim-console/blob/CU-86bywx2j0_SSO-isnt-enabled-on-the-OpenHIM-console-anymore-when-enabled-via-env-vars/config/default-env.json

File: packaging/import-maps-server/Dockerfile-mf [hunk 1]
```diff
@@ -38,4 +38,4 @@ RUN cp update_openhim_urls.sh ${nginxDir}/update_openhim_urls.sh
```
Comment (rcrichton) [inline packaging/import-maps-server/Dockerfile-mf:38]: Looks like the above line need to be updated too.

File: packages/openhim-core-api/src/jembi-openhim-core-api.ts [hunk 2]
```diff
@@ -20,12 +20,39 @@ interface App {
...
+  }
+}
+// Variable to hold the initialized apiClient
+let apiClient = axios.create();
+
+// Call initializeApiClient to setup apiClient before using it
+const initializationPromise = initializeApiClient().catch(console.error);
 
 export async function fetchApps(): Promise<App[]> {
+  await initializationPromise;
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply packages/openhim-core-api/src/jembi-openhim-core-api.ts:51]: This ensures that the right url is set before any of the endpoints can be called.

File: packages/openhim-core-api/src/jembi-openhim-core-api.ts [hunk 3]
```diff
@@ -20,12 +20,39 @@ interface App {
...
+  
+  if (!apiClient) {
+    throw new Error('API client not initialized');
+  }
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply packages/openhim-core-api/src/jembi-openhim-core-api.ts:55]: For consistency I think we should have the check across all functions to ensure that same behavior is cascaded.

**Review by rcrichton — CHANGES_REQUESTED** [review]
Looks good to me, just a final comment I still see a few references to `REACT_APP_OPENHIM_API_BASE_URL` if I do a global search. I think those can be removed now right?

image

**Review by rcrichton — APPROVED** [review]
LGTM

**Pull request merged**